### PR TITLE
#7 Add paywall bloc models pt. 1

### LINKED ISSUE #7: Add the paywall for premium accounts!
# Issue
We want to add a paywall to convert users to premium users! The requirement for this feature is that we can use Google Pay to let the subscriber know of the benefits of going premium
Benefits include:
* Add as many projects as you need
* Right now we cap it at 10 projects per month and 30 projects max for non-paying users
* Tasks are limited to 20 tasks and premium get unlimited tasks
* Premium users get 3 limited mascot choices while paying users get 5

# Note
This ticket can be broken down into multiple tickets

### PR DESCRIPTION
**Pull request by adrdev10**
# What
Paywalls: Added a new firebase model called `Offers` which will serve as the main repo models for 2 different kinds of paying models we want to offer to our users: 2.99 & 4.99

2.99 Offer model -> Will not include AI functionality as previously discussed 
4.99 Offer model -> Will include AI functionality

With this backend-driven approach implementation, we can update our UI for pricing from the Firebase instance. This saves us time because we don't need to rely on constant deploys for ui updates to our pricing models.

- This PR adds the basic bloc functionality in order to fetch data from the Firebase store (`Offer` model). 
- Adds unit test for testing initial state and others 

# Why
In order to implement a paywall feature into the app, we must update the Firebase store model and add the necessary crud operations in the code-base

### CODE REVIEW

File: lib/components/calendar/calendarscreen.dart [hunk 1]
```diff
@@ -4,7 +4,7 @@ import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:intl/intl.dart';
 import 'package:procastiless/components/project/bloc/project_bloc.dart';
 import 'package:procastiless/components/project/bloc/project_state.dart';
-import 'package:procastiless/components/project/data/project.dart';
```
Comment (adrdev10) [inline lib/components/calendar/calendarscreen.dart:7]: @PantherDave  I renamed the folders that store the bloc content's models from `data` to `models` as discussed before.

File: lib/components/project/data/project.dart [hunk 2]
```diff
@@ -1,30 +0,0 @@
-import 'package:cloud_firestore/cloud_firestore.dart';
```
Comment (adrdev10) [inline lib/components/project/data/project.dart]: Not sure why but git thinks I deleted this file but was only changed from one folder to another

File: lib/components/project/data/task.dart [hunk 3]
```diff
@@ -1,41 +0,0 @@
-import 'package:equatable/equatable.dart';
```
Comment (adrdev10) [inline lib/components/project/data/task.dart]: Same with this one

File: lib/components/project/screen/projectscreen.dart [hunk 4]
```diff
@@ -348,7 +348,9 @@ class ProjectScreenState extends State<ProjectScreen> {
                                                       borderRadius:
                                                           BorderRadius.circular(
                                                               10),
-                                                      color: Color(0xffE4C864),
+                                                      color: getPriorityColor(
+                                                          state.projects[i]
```
Comment (adrdev10) [inline lib/components/project/screen/projectscreen.dart:352]: This code is part of a different bug fix which was related to how we display the difficulty tags on the dashboard screen. I just added here to save me some time lol

File: lib/components/dashboard/screen/dashboardscreen.dart [hunk 5]
```diff
@@ -27,80 +29,94 @@ class DashboardState extends State<Dashboard> {
   Widget build(BuildContext context) {
     return BlocBuilder<LoginBloc, LoginState>(
       builder: (context, state) {
-        return Scaffold(
-            floatingActionButton: AddProjectButton(tutorialSettigns),
-            floatingActionButtonLocation:
-                FloatingActionButtonLocation.centerDocked,
-            bottomNavigationBar: Container(
-              decoration: BoxDecoration(
+        if (state is LoggedIn &&
```
Comment (adrdev10) [inline lib/components/dashboard/screen/dashboardscreen.dart:31]: This one was part of the logging issue I fixed the other day. Ignore this part as well

File: lib/components/login/bloc/login_block.dart [hunk 6]
```diff
@@ -18,29 +18,37 @@ class LoginBloc extends Bloc<LoginEvents, LoginState> {
...
+  }
+
   Stream<LoginState> _signInEvent(List<LoginState> states) async* {
     yield InProcessOfLogin();
     trySignInWithAsAnou();
     yield LoggedIn(null);
   }
 
-  Stream<LoginState> _singoutEvent(List<LoginState> states) async* {
+  Stream<LoginState> _singoutEvent() async* {
```
Comment (adrdev10) [inline lib/components/login/bloc/login_block.dart]: At some point we also want to this refactor for the project and task blocks

File: lib/widgets/avatarselection.dart [hunk 7]
```diff
@@ -128,13 +128,13 @@ class AvatarSelectorState extends State<AvatarSelector>
                       if (state is LoggedIn) {
                         if (selectedAvatar == 0) {
                           state.accountUser?.avatarUrl =
-                              'https://firebasestorage.googleapis.com/v0/b/procastiless-6c5f4.appspot.com/o/Group.png?alt=media&token=f5dd18ed-1a3b-403a-aeb0-4db2d1785e3a';
+                              'https://firebasestorage.googleapis.com/v0/b/procastiless.appspot.com/o/Group.png?alt=media&token=d835bde2-e773-4307-8b7b-886d601ba123&_gl=1*ecdhs9*_ga*OTc3MjI4NzQxLjE2OTYzMDUyNDU.*_ga_CW55HF8NVT*MTY5Njk3ODk4Mi42LjEuMTY5Njk3OTYxNi41NC4wLjA.';
```
Comment (adrdev10) [inline lib/widgets/avatarselection.dart:131]: This is part of the change for changing the URL of the avatars

File: lib/components/login/bloc/login_block.dart (hunk 6 above)
Comment (PantherDave) [reply lib/components/login/bloc/login_block.dart]: The method should be named _signOutEvent right?

**Review by PantherDave — APPROVED** [review]

**Pull request merged**